@here/harp-mapview/lib/text/TextStyleCache.ts

```diff
@@ -154,20 +146,20 @@ export class TextStyleCache {
                         textCanvas => textCanvas.fontCatalog.name === style.fontCatalog
                     );
                     style.textCanvas = styledTextCanvas;
-                    style.poiRenderer = defaultPoiRenderer;
                 }
                 if (style.textCanvas === undefined) {
                     if (style.fontCatalog !== undefined) {
                         logger.warn(
-                            `FontCatalog '${style.fontCatalog}' set in TextStyle '${
-                                style.name
-                            }' not found, using default fontCatalog(${
-                                defaultTextCanvas!.fontCatalog.name
-                            }).`
+                            `FontCatalog '${style.fontCatalog}' set in TextStyle '${style.name}'
+                            not found`
+                        );
+                    }
+                    if (textCanvases.length > 0) {
+                        style.textCanvas = textCanvases[0];
+                        logger.info(
+                            `using default fontCatalog(${textCanvases[0].fontCatalog.name}).`
```

Comment:
lines 150-160 : same code as above for default style, please move to free function or lambda.

Reply:
done